validators/cgroup_validator.go

```diff
@@ -82,6 +85,31 @@ func (c *CgroupsValidator) validateCgroupSubsystems(cgroups, subsystems []string
 }
 
 func (c *CgroupsValidator) getCgroupSubsystems() ([]string, error) {
+	// Get the subsystems from /sys/fs/cgroup/cgroup.controllers when cgroup v2 is used.
+	// /proc/cgroups is meaningless for v2
+	// https://github.com/torvalds/linux/blob/v5.3/Documentation/admin-guide/cgroup-v2.rst#deprecated-v1-core-features
+	var st unix.Statfs_t
+	if err := unix.Statfs(unifiedMountpoint, &st); err != nil {
+		return []string{}, errors.Wrap(err, "cannot statfs the cgroupv2 root")
+	}
+	if st.Type == unix.CGROUP2_SUPER_MAGIC {
+		// Some controllers are implicitly enabled by the kernel.
+		// Those controllers do not appear in /sys/fs/cgroup/cgroup.controllers.
+		// https://github.com/torvalds/linux/blob/v5.3/kernel/cgroup/cgroup.c#L433-L434
+		// We assume these are always available, as it is hard to detect availability.
+		// So, we hardcode the following as "pseudo" controllers.
+		// - devices: implemented in kernel 4.15
+		// - freezer: implemented in kernel 5.2
+		pseudo := []string{"devices", "freezer"}
+		data, err := ioutil.ReadFile("/sys/fs/cgroup/cgroup.controllers")
```

Comment:
nit: This can reuse the `unifiedMountpoint` constant.

Reply:
I reused unifiedMountpoint.  Thanks!